src/lib/openssl_crypto.c

```diff
@@ -95,6 +95,12 @@
 
 #include <botan/ffi.h>
 
+#define FAIL(str)  \
```

Comment:
Could you please explain the the reason for wrapping these statements in `do {} while (false)` and perhaps comment it?